twill-core/src/main/java/org/apache/twill/internal/TwillRuntimeSpecification.java

```diff
@@ -105,4 +114,17 @@ public String getRmSchedulerAddr() {
   public Map<String, Integer> getMaxRetries() {
     return maxRetries;
   }
+
+  /**
+   * Returns the ZK connection string for the Kafka used for log collections,
+   * or {@code null} if log collection is disabled.
+   */
+  @Nullable
+  public String getKafkaZKConnect() {
+    if (!isLogCollectionEnabled()) {
```

Comment:
I am not too keen about having this "side effect" check for log collection enabled here to return KafkaZKConnect.
Caller can always check `isLogCollectionEnabled()` to figure out whether log collection is disabled, right?